internal/kafka/internal/services/clusters_test.go

```diff
@@ -2340,3 +2475,406 @@ func Test_ClusterService_GetExternalID(t *testing.T) {
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := clusterService{
+				connectionFactory: tt.fields.connectionFactory,
+				providerFactory:   tt.fields.providerFactory,
+			}
+			got, err := c.IsStrimziKafkaVersionAvailableInCluster(tt.args.cluster, tt.args.strimziVersion, tt.args.kafkaVersion, tt.args.ibpVersion)
+			Expect(got).To(Equal(tt.want))
+			Expect(err != nil).To(Equal(tt.wantErr))
+		})
+	}
+}
+
+func Test_clusterService_SetComputeNodes(t *testing.T) {
+	type fields struct {
+		connectionFactory *db.ConnectionFactory
+		providerFactory   clusters.ProviderFactory
+	}
+	type args struct {
+		clusterID string
+		numNodes  int
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    *types.ClusterSpec
+		wantErr bool
+		setupFn func()
+	}{
+		{
+			name: "should return nil and error if clusterID is undefined",
+			fields: fields{
+				connectionFactory: db.NewMockConnectionFactory(nil),
+				providerFactory:   &clusters.ProviderFactoryMock{},
+			},
+			args:    args{},
+			want:    nil,
+			wantErr: true,
+			setupFn: func() {},
+		},
+		{
+			name: "Should return the cluster after setting compute nodes",
+			fields: fields{
+				connectionFactory: db.NewMockConnectionFactory(nil),
+				providerFactory: &clusters.ProviderFactoryMock{
+					GetProviderFunc: func(providerType api.ClusterProviderType) (clusters.Provider, error) {
+						return &clusters.ProviderMock{
+							CreateFunc: func(request *types.ClusterRequest) (*types.ClusterSpec, error) {
+								return &types.ClusterSpec{
+									InternalID: testClusterID,
+								}, nil
+							},
+							SetComputeNodesFunc: func(clusterSpec *types.ClusterSpec, numNodes int) (*types.ClusterSpec, error) {
+								return &types.ClusterSpec{
+									InternalID: testClusterID,
+								}, nil
+							},
+						}, nil
+
+					}},
+			},
+			args: args{
+				clusterID: testClusterID,
+				numNodes:  2,
+			},
+			setupFn: func() {
+				res := []map[string]interface{}{
+					{
+						"id":          "testid",
+						"cluster_id":  testClusterID,
+						"cluster_dns": "",
+						"client_id":   "ClientID",
+					},
+				}
+				mocket.Catcher.Reset().
+					NewMock().WithQuery(`SELECT * FROM "clusters" WHERE "clusters"."cluster_id" = $1`).
+					WithArgs(testClusterID).
+					WithReply(res)
+			},
+			want: &types.ClusterSpec{
+				InternalID: testClusterID,
+			},
+			wantErr: false,
+		},
+		{
+			name: "failed to get provider implementation",
+			fields: fields{
+				connectionFactory: db.NewMockConnectionFactory(nil),
+				providerFactory: &clusters.ProviderFactoryMock{
+					GetProviderFunc: func(providerType api.ClusterProviderType) (clusters.Provider, error) {
+						return nil, &apiErrors.ServiceError{}
+					},
+				},
+			},
+			args: args{
+				clusterID: testClusterID,
+				numNodes:  2,
+			},
+			want:    nil,
+			wantErr: true,
+			setupFn: func() {},
+		},
+		{
+			name: "failed to get compute nodes info from provider",
```

Comment:
This test seems the same as https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/981/files#r862991938. This is, both seem to test what happens when the call to the `SetupComputeNodes` of the OCM client fails